src/prefect/orion/orchestration/global_policy.py

```diff
@@ -78,7 +79,8 @@ class SetRunStateName(BaseUniversalTransform):
     async def before_transition(self, context: OrchestrationContext) -> None:
 
         # record the new state's name
-        context.run.state_name = context.proposed_state.name
+        if not self.nullified_transition():
+            context.run.state_name = context.proposed_state.name
```

Comment:
I'm more into:

```
if self.nullified_transition():
    return

...do the things...
```

Just because it keeps the indentation lower and it's a little clearer that the check is a guard for the whole function in a lot of these cases